[c10d] Add comments to the rest environment variable within NCCLPG

fduwjj opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #117168
* #117093
* __->__ #117092
* #116545

Not every environment within NCCLPG has comments, let's add comments to each of them.


cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @wz337 @tianyu-l @wconstab @yf225

fduwjj committed:
[c10d] Add comments to the rest environment variable within NCCLPG

[ghstack-poisoned]

wconstab commented on torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp:
```diff
@@ -59,16 +59,24 @@ static std::vector<std::string> TORCH_NCCL_DESYNC_DEBUG = {
     "TORCH_NCCL_DESYNC_DEBUG",
     "NCCL_DESYNC_DEBUG"};
 
+// Environment variable to enable whether to record and check the start
```
can remove the "Environment variable" prefix from all of these

"of one collective" (?)  not "all collectives"? 

How about
"Enable recording start-events for all ProcessGroupNCCL collectives, and compute accurate collective timing per-collective.  (Note: end-events are recorded by default).  Can increase chances of a watchdog hang due to querying cudaEventElapsedTime() API"

fduwjj replied:
Maybe I should say each collective but using "all" also works.

wconstab commented on torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp:
```diff
@@ -59,16 +59,24 @@ static std::vector<std::string> TORCH_NCCL_DESYNC_DEBUG = {
     "NCCL_ENABLE_TIMING"};
 
+// Environment variable to enable monitoring thread which abort the process
+// when no heartbeat is detected after TORCH_NCCL_HEARTBEAT_TIMEOUT_SEC.
```
"Enable monitoring thread which aborts the process when the ProcessGroupNCCL Watchdog thread gets stuck, which can happen due to calling CUDA/NCCL APIs that may hang.  Useful to prevent jobs being stuck  for a longer time than necessary tying up cluster resources."

wconstab commented on torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp:
```diff
@@ -59,16 +59,24 @@ static std::vector<std::string> TORCH_NCCL_DESYNC_DEBUG = {
     "TORCH_NCCL_HEARTBEAT_TIMEOUT_SEC"};
 
+// Environment variable to configure the maximum number of collectives that we
+// cache the flight recorder for in the global ring buffer.
```
"The number of events we store in the flight recorder's ring buffer.  (One event could be the start or end of a collective, for example)"

fduwjj committed:
Update on "[c10d] Add comments to the rest environment variable within NCCLPG"




cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

fduwjj committed:
Update on "[c10d] Add comments to the rest environment variable within NCCLPG"


Not every environment within NCCLPG has comments, let's add comments to each of them.


cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

kwen2501 reviewed (approved):
LGTM!

fduwjj commented:
@pytorchbot merge
